packages/warehouses/src/warehouseClients/WarehouseBaseClient.ts

```diff
@@ -39,13 +39,40 @@ export default class WarehouseBaseClient<T extends CreateWarehouseCredentials>
         throw new Error('Warehouse method not implemented.');
     }
 
-    async runQuery(sql: string): Promise<{
-        fields: Record<string, { type: DimensionType }>;
-        rows: Record<string, any>[];
-    }> {
+    async streamQuery(
+        query: string,
+        streamCallback: (data: WarehouseResults) => void,
+        options: {
+            tags?: Record<string, string>;
+            timezone?: string;
+        },
+    ): Promise<void> {
         throw new Error('Warehouse method not implemented.');
     }
 
+    async runQuery(
+        sql: string,
+        tags?: Record<string, string>,
+        timezone?: string,
+    ) {
+        let fields: WarehouseResults['fields'] = {};
+        const rows: WarehouseResults['rows'] = [];
+
+        await this.streamQuery(
+            sql,
+            (data) => {
+                fields = data.fields;
```

Comment:
so you're overwritting fields with the last chunk of data. Is this compatible with all warehouses ?

Reply:
Yes 👍 I opted to send them in all callback calls to be consistent even if the value is the same. In most warehouse clients we compute the `fields` only once at the start.